source/app/util/environment-check.ts

```diff
@@ -130,5 +132,59 @@ export default async function environmentCheck (): Promise<void> {
     }
   }
 
+  try {
+    process.env.ZETTLR_IS_TRAY_SUPPORTED = await isTraySupported() ? '1' : '0'
+  } catch (err) {
+    process.env.ZETTLR_IS_TRAY_SUPPORTED = '0'
+    process.env.ZETTLR_TRAY_ERROR = err.message
+    global.log.warning(err.message)
+  }
+
   global.log.info('Environment check complete.')
 }
+
+/**
+ * Check if system supports a Tray.
+ *
+ * Returns true on Windows and MacOS.
+ * Returns true on Linux with Gnome desktop if Gnome Extension
+ * 'KStatusNotifierItem/AppIndicator Support' is installed, otherwise throws
+ * an {Error} with the reason why.
+ * Returns true on Linux with other desktops. e.g. KDE, XFCE, etc.
+ *
+ * @return {*}  {Promise<boolean>} If supported, returns true. Never returns
+ *              false.
+ * @throws {Error} Details why the Tray is not supported; or
+ *                 Details the error if an error occurred while checking Tray
+ *                 support.
+ */
+export async function isTraySupported (): Promise<boolean> {
```

Comment:
I'm currently thinking whether we should put the function into its own file …? Mh, yeah, makes sense. (Btw in this case you don't need to export it, since it's only being used in this file ;)